tracing-appender/src/rolling.rs

```diff
@@ -437,67 +501,117 @@ pub fn never(directory: impl AsRef<Path>, file_name: impl AsRef<Path>) -> Rollin
 /// # }
 /// ```
 #[derive(Clone, Eq, PartialEq, Debug)]
-pub struct Rotation(RotationKind);
+pub struct Rotation {
+    timed: Timed,
+    max_bytes: Option<u64>,
+}
 
-#[derive(Clone, Eq, PartialEq, Debug)]
-enum RotationKind {
+#[derive(Clone, Copy, Eq, PartialEq, Debug)]
+enum Timed {
     Minutely,
     Hourly,
     Daily,
     Never,
 }
 
 impl Rotation {
-    /// Provides an minutely rotation
-    pub const MINUTELY: Self = Self(RotationKind::Minutely);
-    /// Provides an hourly rotation
-    pub const HOURLY: Self = Self(RotationKind::Hourly);
-    /// Provides a daily rotation
-    pub const DAILY: Self = Self(RotationKind::Daily);
+    /// Provides a minutely rotation.
+    pub const MINUTELY: Self = Self {
+        timed: Timed::Minutely,
+        max_bytes: None,
+    };
+
+    /// Provides an hourly rotation.
+    pub const HOURLY: Self = Self {
+        timed: Timed::Hourly,
+        max_bytes: None,
+    };
+
+    /// Provides a daily rotation.
+    pub const DAILY: Self = Self {
+        timed: Timed::Daily,
+        max_bytes: None,
+    };
+
     /// Provides a rotation that never rotates.
-    pub const NEVER: Self = Self(RotationKind::Never);
+    pub const NEVER: Self = Self {
+        timed: Timed::Never,
+        max_bytes: None,
+    };
+
+    /// Provides a size-based rotation.
+    pub const fn max_bytes(number_of_bytes: u64) -> Self {
+        Self {
+            timed: Timed::Never,
+            max_bytes: Some(number_of_bytes),
+        }
+    }
+
+    /// Provides a minutely and size-based rotation.
+    pub const fn minutely_with_max_bytes(number_of_bytes: u64) -> Self {
+        Self {
+            timed: Timed::Minutely,
+            max_bytes: Some(number_of_bytes),
+        }
+    }
+
+    /// Provides a hourly and size-based rotation.
+    pub const fn hourly_with_max_bytes(number_of_bytes: u64) -> Self {
+        Self {
+            timed: Timed::Hourly,
+            max_bytes: Some(number_of_bytes),
+        }
+    }
+
+    /// Provides a daily and size-based rotation.
+    pub const fn daily_with_max_bytes(number_of_bytes: u64) -> Self {
+        Self {
+            timed: Timed::Daily,
+            max_bytes: Some(number_of_bytes),
+        }
```

Comment:
to reduce the total number of functions in the API, what do you think about just having a `.with_max_bytes` function taking an existing `Rotation` and adding a byte limit? so, 
```suggestion
    /// Adds a maximum size limit to an existing `Rotation`.
    ///
    /// The new `Rotation` will rotate the log file when the log file reaches
    /// the specified size limit, or when the rotation period elapses, whichever 
    /// occurs first.
    pub const fn with_max_bytes(self, number_of_bytes: u64) -> Self {
        Self {
            max_bytes: Some(number_of_bytes),
            ..self
        }
    } 
```

Reply:
This sounds good to me!